octoprint/datadog_checks/octoprint/check.py

```diff
@@ -0,0 +1,138 @@
+        req = requests.get(url, timeout=timeout, headers=headers)
+        return req.json()
+
+    def get_bed_temp(self, server, key, timeout):
+        PATH = "/api/printer/bed"
+        url = server + PATH
+        headers = {"X-Api-Key": key, "content-type": "application/json"}
+        req = requests.get(url, timeout=timeout, headers=headers)
+        return req.json()
+
+    def check(self, instance):
+        octo_api_key = instance.get('octo_api_key')
+        self.log.debug('octo_api_key (from config): %s', octo_api_key)
+        self.log.debug('octo_server: %s', SERVER)
+        rpi_core_temp = self.get_rpi_core_temp()
+        self.gauge("octoprint.rpi_core_temp", rpi_core_temp)
+
+        # get api data
+        job_info = self.get_job_info(SERVER, octo_api_key, TIMEOUT)
+
+        # # Job State
+        state = job_info["state"]
+        # States: Printing, Paused, Cancelled, Operational...
+        # The proper thing would be a a case/switch statement
+        if state == "Operational":
+            printer_state = 0
+        elif state == "Paused":
+            printer_state = 1
+        elif state == "Printing":
+            printer_state = 2
+        else:
+            printer_state = -1
+        self.gauge("octoprint.printer_state", printer_state)
+
+        # Print Job Percent Completed and Time Estimate
+        est_print_time = self.seconds_to_minutes(job_info["job"]["estimatedPrintTime"])
+        pct_completed = job_info["progress"]["completion"]
+        self.gauge("octoprint.est_print_time", est_print_time)
+        self.gauge("octoprint.pct_completed", pct_completed)
+
+        # Print Job Elapsed and Remaining Times
+        print_job_time = self.seconds_to_minutes(job_info["progress"]["printTime"])
+        print_job_time_left = self.seconds_to_minutes(job_info["progress"]["printTimeLeft"])
+        self.gauge("octoprint.print_job_time", print_job_time)
+        self.gauge("octoprint.print_job_time_left", print_job_time_left)
+
+        # # Extruder Temperatures
+        extruder_temps = self.get_tool_temp(SERVER, octo_api_key, TIMEOUT)
+        for key in extruder_temps.keys():
+            toolname = key
+            current_tool_temp = extruder_temps[toolname]["actual"]
+            target_tool_temp = extruder_temps[toolname]["target"]
+            self.gauge("octoprint." + toolname + ".current_tool_temp", current_tool_temp)
+            self.gauge("octoprint." + toolname + ".target_tool_temp", target_tool_temp)
+
+        # # Bed Temperatures
+        bed_temp = self.get_bed_temp(SERVER, octo_api_key, TIMEOUT)
+        for key in bed_temp.keys():
+            bedname = key
+            current_bed_temp = bed_temp[bedname]["actual"]
+            target_bed_temp = bed_temp[bedname]["target"]
+            self.gauge("octoprint." + bedname + ".current_bed_temp", current_bed_temp)
+            self.gauge("octoprint." + bedname + ".target_bed_temp", target_bed_temp)
```

Comment:
Similar comment: use consistent metric, and move `bedname` to a tag.

Reply:
As we talked about, I'll implement this at a later time.